frameit/lib/frameit/editor.rb

```diff
@@ -347,10 +350,64 @@ def build_title_images(max_width, max_height)
           i.interline_spacing interline_spacing if interline_spacing
           i.fill fetch_config[key.to_s]['color']
         end
-        title_image.trim # remove white space
+
+        # Trimming the image will result in the loss of the common baseline between the text in all images.
+        # title_image.trim # remove white space
 
         results[key] = title_image
+
+        # Hence retrieve the calculated trim bounding box:
+        # CALCULATED: trim bounding box (without actually trimming), see: http://www.imagemagick.org/script/escape.php
+        calculated_trim_box = title_image.identify { |b| b.format("%@") }
+
+        # Get vertical top offset from the trim box by converting it from string to an integer array:
+        trim_values = calculated_trim_box.split(/[\sx+]/).map(&:to_i) # The format is "wxh+X+Y", so use multiple string separators: the whitespace "\s", "x" and "+"
+
+        # Get the minimum top offset of the trim box:
+        if trim_values[3] < top_vertical_trim_offset
```

Comment:
This seems a bit magical to know that we should index in to `3` and `1` below, is there any explanation we could add here to make this a bit more clear?

Reply:
What about this extended comment:
```Ruby
# The .identify with format("%@") returns a string with "<width>x<height>+<offset_x>+<offset_y>". Extract these 4 parameters into an integer array, by using multiple string separators: the whitespace "\s", "x" and "+".
```

To also answer your next comment. Sure, we can define these indexes.
Do you have any naming convention for these kind of constants?
Example:
```Ruby
index_trim_box_width = 0
index_trim_box_height = 1
index_trim_box_offset_x = 2
index_trim_box_offset_y = 3
```

Reply:
I think code that is self explanatory is better than comments, I would say that you could do something like: 

```
trim_box_width = trim_values[0]
trim_box_height = trim_values[1]
trim_box_offset_x = trim_values[2]
trim_box_offset_y = trim_values[3]
```

Reply:
Understood.

Reply:
Basically only need trim_box_height and trim_box_offset_y, this saves 2 lines already 👊

Reply:
Hmm, unfortunately I have to define them twice, because the trim box is used in 2 loops:
1. The loop where the image is created. To get the minimum top and maximum bottom offset of all images.
2. The loop where the trim boxes are compensated based on the earlier minimum top and maximum bottom offsets, to maintain the text baseline.

So now I have this with 2 times `# Get the height and vertical offset`:
```Ruby
        # The .identify with format("%@") returns a string with "<width>x<height>+<offset_x>+<offset_y>". Extract these 4 parameters into an integer array, by using multiple string separators: the whitespace "\s", "x" and "+":
        trim_values = calculated_trim_box.split(/[\sx+]/).map(&:to_i)
                            
        # Get the height and vertical offset:
        trim_box_height = trim_values[1]
        trim_box_offset_y = trim_values[3]

        # Get the minimum top offset of the trim box:
        if trim_box_offset_y < top_vertical_trim_offset
          top_vertical_trim_offset = trim_box_offset_y
        end

        # Get the maximum bottom offset of the trimbox, this is the top offset + height:
        if (trim_box_offset_y + trim_box_height) > bottom_vertical_trim_offset
          bottom_vertical_trim_offset = trim_box_offset_y + trim_box_height
        end

        # Store for the crop action:
        trim_boxes[key] = trim_values
      end

      # Crop images based on top_vertical_trim_offset to maintain text baseline:
      words.each do |key|
        # Get matching trim box:
        trim_box = trim_boxes[key]

        # Get the height and vertical offset:
        trim_box_height = trim_box[1]
        trim_box_offset_y = trim_box[3]
```

Reply:
Okay. So we have a couple options. I don't want you to keep having to make changes as you've already put in a great deal of work here. I am cool to merge it as is, and we can go back in and clean it up later.

The other option would be to create some sort of Ruby object that encapsulates these box dimensions and then have an initializer where it could be created from the array of `trim_boxes` that way it could be created and used a lot more efficiently *and* have the extra benefit of being abstracted into another class and therefor maybe help with the 320 line restriction.

What are your thoughts?

Reply:
Option 2 is definitely preferred, as it improves on 3 subjects:
- readability
- 320 line check
- re-use

It would be great if you can make it, as I lack the Ruby skills. Or show me a similar file in the repo that I can use as example.

My functional thoughts:
1. The initializer uses the raw string from the .identify as input and retrieves it's 4 contents to create the trim box object with 4 populated properties: width, height, offset_x and offset_y
2. The current Class can read and change these object properties 
3. A method that returns the string format for the `.crop` method.

Reply:
Yeah that sounds awesome and we are definitely happy to help with that!

Reply:
For me just the outline of a class with an init method, a property and 1 method that returns a string is enough. Then I can populate it all myself.

What do you prefer as Class name?
Although currently everything is titled "trim box", maybe "crop box", "crop frame" or "crop canvas" are more appropriate?